---
layout: page
title: Bioinformatics Experience
subtitle: R (Fluent) | Python (Intermediate)
---

<style>
  body {
    background-color: #254E58;
    color: #f7f9fa;
  }
  a {
    color: #02b8de;
    text-decoration: none;
    border-bottom: 1px solid transparent;
    transition: border-bottom 0.2s ease;
  }
  a:hover {
    border-bottom: 1px solid #6E6658;
  }
  details {
    border: 1px solid #f7f9fa;
    border-radius: 6px;
    padding: 1rem 1.5rem;
    margin-bottom: 1rem;
    background: #254E58;
    transition: background 0.3s ease;
  }
  details[open] {
    background: #112D32;
  }
  summary {
    font-size: 1.3rem;
    font-weight: 600;
    cursor: pointer;
    list-style: none;
    color: #f7f9fa;
  }
  summary::-webkit-details-marker {
    display: none;
  }
  summary:before {
    content: "\f054"; /* FontAwesome caret-right */
    font-family: "Font Awesome 5 Free";
    font-weight: 900;
    display: inline-block;
    margin-right: 10px;
    transition: transform 0.3s ease;
    color: #02b8de;
  }
  details[open] summary:before {
    transform: rotate(90deg);
  }
  ul {
    padding-left: 1.2rem;
    margin-top: 0.5rem;
  }
  ul li {
    margin-bottom: 0.5rem;
  }
  .summary-icon {
    margin-right: 10px;
    color: #02b8de;
  }
</style>

<details >
  <summary><i class="fas fa-server summary-icon"></i>Bioinformatics Service</summary>
  <p>
    I established and lead the phenotypic screening facility’s bioinformatics service, providing bespoke computational support to academic and industry users. This chargeable service supports a wide range of life science disciplines, including cancer biology, immunology, neuroscience, developmental biology, and skin research.
  </p>
  <p>
    I specialise in deriving insights from complex imaging and omics datasets and in developing scalable, reproducible analysis pipelines to advance cutting-edge research.
  </p>
  <p>
    I also promote best practice and version control through the <a href="https://github.com/Phenotypic-Screening-QMUL" target="_blank">facility GitHub</a>.
  </p>
</details>

<details>
  <summary><i class="fas fa-robot summary-icon"></i>Machine Learning (ML) and Artificial Intelligence (AI)</summary>
  <p>
  <strong>Techniques:</strong>
  <ul>
    <li><strong>Supervised learning:</strong> Logistic regression, Lasso regularisation, Elastic Net, Support Vector Machines (SVM), Random Forest, Multiple Discriminant Analysis, Neural Networks</li>
    <li><strong>Unsupervised learning:</strong> Hierarchical clustering, k-means clustering, Principal Component Analysis (PCA), Exploratory Factor Analysis (EFA)</li>
  </ul>
  <strong>Projects:</strong>
  <ul>
    <li><strong>SAMP-Score:</strong> Ensemble machine learning model for senescence detection to support drug discovery in cancer - <a href="https://ryanjwallis.github.io/2024-02-07-SAMP-Score/" target="_blank">Project Link</a></li>
    <li><strong>SenPred:</strong> Senescence classification model for single-cell RNA sequencing transcriptomic data developed from 3D <em>in vitro</em> cell culture models - <a href="https://ryanjwallis.github.io/2023-10-23-SenPred/" target="_blank">Project Link</a></li>
    <li>Prognostic multiplexed immunofluorescence model utilising the spatial assessment in oral epithelial dysplasia (<em>manuscript in preparation</em>)</li>
  </ul>
  <strong>ML Reviews:</strong>
  <ul>
    <li><a href="https://pubmed.ncbi.nlm.nih.gov/39187617/" target="_blank">ML assessment of senescence morphology (PubMed ID: 39187617)</a></li>
    <li><a href="https://pubmed.ncbi.nlm.nih.gov/37016180/" target="_blank">ML techniques in senescence research (PubMed ID: 37016180)</a></li>
  </ul>
</details>

<details>
  <summary><i class="fas fa-image summary-icon"></i>High-Content Image Analysis</summary>
  <p>
  <strong>Techniques:</strong>
  <ul>
    <li><strong>Analysis:</strong> Z-Scores, cluster-analysis, IC50s, batch normalisation</li>
    <li><strong>Visualisations:</strong> Heatmaps, frequency distributions, dimensionality reduction (UMAP/TSNE)</li>
  </ul>
  <strong>Projects:</strong>
  <ul>
    <li>Unsupervised characterisation of senescence via phenotypic assessment of morphology - <a href="https://ryanjwallis.github.io/22022-05-16-SAMPs/" target="_blank">Project Link</a></li>
    <li>Phenocopying methodology comparing a genome-wide siRNA screen to a novel compound for target identification (<em>in evaluation phase with commercial partner</em>) - <a href="https://www.qmul.ac.uk/blizard/about/news/items/queen-mary-research-team-heads-discovery-of-a-new-method-to-stop-the-growth-of-cancer-cells.html" target="_blank">Press Release</a></li>
    <li>Developed reusable pipelines and HTML guides for users to standardise imaging data analysis - <a href="https://github.com/Phenotypic-Screening-QMUL/InCarta-Scripts" target="_blank">GitHub</a></li>
  </ul>
  <strong>HCA Reviews:</strong>
  <ul>
    <li><a href="https://pubmed.ncbi.nlm.nih.gov/37793342/" target="_blank">CRISPR Screening in Senescence Drug Discovery (PubMed ID: 37793342)</a></li>
  </ul>
</details>

<details>
  <summary><i class="fas fa-map-marker-alt summary-icon"></i>Spatial Biology</summary>
  <p>Develop bespoke and sophisticated analysis pipelines for users of our Cell DIVE and Phenocycler Fusion multiplex immunofluorescence platforms via the HALO digital pathology software. I have also datamined published spatial transcriptomics datasets for users as a service.</p>
  <strong>Techniques:</strong>
  <ul>
    <li>Cluster-based cell phenotyping utilising dimensionality reduction and subject knowledge to assign cell types</li>
    <li>K-nearest neighbour analysis to determine cell type tissue distribution</li>
    <li>Cell Neighbourhood analysis to determine cell type enrichment within tissue</li>
    <li>Spatial distance and proximity assessments between cells and tissue architecture</li>
  </ul>
</details>

<details>
  <summary><i class="fas fa-server summary-icon"></i>High Performance Computing</summary>
  <p>I have experience analysing both my own and user projects through the university's High Performance Computing (<a href="https://docs.hpc.qmul.ac.uk/" target="_blank">HPC</a>) cluster via both a web interface and the command line.</p>
</details>

<details>
  <summary><i class="fas fa-dna summary-icon"></i>Genomic / Proteomic Analysis</summary>
   <p>
  <strong>Projects:</strong>
  <ul>
    <li>Proteomic assessment of mass spectrometry data from exosomes and conditioned media - <a href="https://ryanjwallis.github.io/2021-02-18-Senescent-Exosomes/" target="_blank">Project Link</a></li>
    <li>sc-RNAseq transcriptomics analysis using example datasets from satijalab - <a href="https://ryanjwallis.github.io/22023-10-01-Seurat-scRNAseq-Transcriptomics/" target="_blank">Project Link</a></li>
  </ul>
</details>

<details>
  <summary><i class="fas fa-graduation-cap summary-icon"></i>Formal Courses</summary>
  <p>
  <ul>
    <li>DataCamp “Data Scientist with R” track (88-hour online course).</li>
    <li>Biochemical Society “R for Biochemists 101” (5-week online introduction to R).</li>
    <li>LinkedIn Learning “Become a Data Scientist” (17-hour introductory course).</li>
    <li>DataCamp “Data Analyst with Python” track (36-hour online course).</li>
    <li>DataCamp "Machine Learning Fundamentals with Python" track (16-hour online course).</li>
  </ul>
</details>
