---
layout: page
title: Bioinformatics Experience
subtitle: R (Fluent) | Python (Intermediate)
---

### Bioinformatics Service
I established and lead the phenotypic screening facility’s bioinformatics service, providing bespoke computational support to academic and industry users. This chargeable service supports a wide range of life science disciplines, including cancer biology, immunology, neuroscience, developmental biology, and skin research.

I specialise in deriving insights from complex imaging and omics datasets and in developing scalable, reproducible analysis pipelines to advance cutting-edge research.

I also promote best practice and version control through the <a href="https://github.com/Phenotypic-Screening-QMUL" target="_blank"> facility github</a>

### Machine Learning (ML) and Artificial Intelligence (AI)
I have a particular interest in applying machine learning (ML) and artificial intelligence (AI) to solve complex biological problems and have successfully implemented these approaches across diverse data types.

#### Techniques:
**Supervised learning**: Logistic regression, Lasso regularisation, Elastic Net, Support Vector Machines (SVM), Random Forest, Multiple Discriminant Analysis, Neural Networks<br>
**Unsupervised learning**: Hierarchical clustering, k-means clustering, Principal Component Analysis (PCA), Exploratory Factor Analysis (EFA)

#### Projects:
- **SAMP-Score**: Ensemble machine learning model for senescence detection to support drug discovery in cancer - <a href="https://ryanjwallis.github.io/2024-02-07-SAMP-Score/" target="_blank">Project Link</a> 

- **SenPred**: Senescence classification model for single-cell RNA sequencing transcriptomic data developed from 3D _in vitro_ cell culture models - <a href="https://ryanjwallis.github.io/2023-10-23-SenPred/" target="_blank">Project Link</a>

- Prognostic multiplexed immunofluorescence model utilising the spatial assessment in oral epithelial dysplasia (_manuscript in preparation_)

#### Topic Reviews
<a href="https://pubmed.ncbi.nlm.nih.gov/39187617/" target="_blank"> ML assessment of senescence morphology (PubMed ID: 39187617)</a><br>
<a href="https://pubmed.ncbi.nlm.nih.gov/37016180/" target="_blank"> ML techniques in senescence research (PubMed ID: 37016180)</a>

### High-Content Image Analysis
I have extensive experience analysing large-scale imaging data from both compound and genetic screens, as well as developing tools for drug discovery. 

#### Techniques
**Analysis**: Z-Scores, cluster-analysis, IC50s, batch normalisation <br>
**Visualisations**: Heatmaps, frequency distributions, dimensionality reduction (UMAP/TSNE)

#### Projects:
- Unsupervised characterisation of senescence via phenotypic assessment of morphology - <a href="https://ryanjwallis.github.io/22022-05-16-SAMPs/" target="_blank">Project Link</a> <br>
- Phenocopying methodology comparing a genome-wide siRNA screen to  a novel compound for target identification (_in evaluation phase with commercial partner_) - <a href="https://www.qmul.ac.uk/blizard/about/news/items/queen-mary-research-team-heads-discovery-of-a-new-method-to-stop-the-growth-of-cancer-cells.html" target="_blank"> Press Release </a> <br>
- Developed a range of reusable pipelines and workflows, as well as HTML guides for users to standardise the analysis of imaging data  <a href="https://github.com/Phenotypic-Screening-QMUL/InCarta-Scripts" target="_blank"> GitHub </a>

#### Topic Reviews
<a href="https://pubmed.ncbi.nlm.nih.gov/37793342/" target="_blank"> CRISPR Screening in Senescence Drug Discovery (PubMed ID: 37793342)</a>

### Spatial Biology



### High Performance Computing
I have experience analysing both my own and user projects through the university's High Performance Computing (<a href="https://docs.hpc.qmul.ac.uk/" target="_blank"> HPC </a>) cluster via both a web interface and the command line 

### "-omics" Analysis
I have experience analysing a range of other "-omics" datasets.

#### Projects:
- Proteomic assessment of mass spectrometry data from exosomes and conditioned media - <a href="https://ryanjwallis.github.io/2021-02-18-Senescent-Exosomes/" target="_blank">Project Link</a><br>
- sc-RNAseq transcriptomics analysis using example datasets from satijalab - <a href="https://ryanjwallis.github.io/22023-10-01-Seurat-scRNAseq-Transcriptomics/" target="_blank">Project Link</a>

### Formal Courses
− DataCamp “Data Scientist with R” track (88-hour online course).<br>
− Biochemical Society “R for Biochemists 101” (5-week online introduction to R).<br>
− LinkedIn Learning “Become a Data Scientist” (17-hour introductory course).<br>
− DataCamp “Data Analyst with Python” track (36-hour online course).<br>
− DataCamp "Machine Learning Fundamentals with Python" track (16-hour online course).







